lib/improver/orographic_enhancement.py

```diff
@@ -0,0 +1,600 @@
+
+        # generate 3D array of distances to source points
+        distance_weight = self._get_distance_weights(wind_speed, max_sin_cos)
+
+        # calculate positions of source points
+        x_source, y_source = self._locate_source_points(
+            wind_speed, distance_weight, sin_wind_dir, cos_wind_dir)
+
+        # compute weighted enhancements summed over all source points
+        orogenh, sum_of_weights = self._compute_weighted_values(
+            site_orogenh, x_source, y_source, distance_weight, wind_speed)
+
+        # normalise by weights and scale by efficiency factor
+        orogenh[~mask] = self.efficiency_factor * np.divide(
+            orogenh[~mask], sum_of_weights[~mask])
+
+        return orogenh
+
+    def _create_output_cubes(self, orogenh_data, reference_cube):
+        """
+        Create two output cubes of orographic enhancement on different grids
+
+        Args:
+            orogenh_data (np.ndarray):
+                Orographic enhancement value in mm h-1
+            reference_cube (iris.cube.Cube):
+                Cube with the correct time and forecast period coordinates on
+                the UK 2 km standard grid
+
+        Returns:
+            (tuple): tuple containing:
+                **orogenh** (iris.cube.Cube):
+                    Orographic enhancement cube on 1 km UKPP grid
+                **orogenh_standard_grid** (iris.cube.Cube):
+                    Orographic enhancement cube on 2 km standard grid, padded
+                    with masked np.nans
+        """
+        # create cube containing high resolution data in mm/h
+        x_coord = self.topography.coord(axis='x')
+        y_coord = self.topography.coord(axis='y')
+        attributes = {'institution': 'Met Office', 'source': 'IMPROVER'}
+        orogenh = iris.cube.Cube(
+            orogenh_data, long_name="orographic_enhancement",
+            units="mm h-1", attributes=attributes,
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+        orogenh.add_aux_coord(reference_cube.coord('time'))
+        orogenh.add_aux_coord(reference_cube.coord('forecast_reference_time'))
+        orogenh.add_aux_coord(reference_cube.coord('forecast_period'))
+
+        # regrid the orographic enhancement cube onto the standard grid and
+        # mask extrapolated points
+        orogenh_standard_grid = orogenh.regrid(
+            reference_cube, iris.analysis.Linear(extrapolation_mode='mask'))
+
+        for axis in ['x', 'y']:
+            orogenh_standard_grid = sort_coord_in_cube(
+                orogenh_standard_grid, orogenh_standard_grid.coord(axis=axis))
+
+        return orogenh, orogenh_standard_grid
+
+    def process(self, temperature, humidity, pressure, uwind, vwind,
+                topography, svp_method='IMPROVER'):
+        """
+        Calculate precipitation enhancement over orography on standard (2 km)
+        and high resolution (1 km UKPP domain) grids
```

Comment:
As I understand it, there is no reason for these to be 2km and 1km so we should refer to them as standard and high-resolution throughout and leave it at that. I suggest inserting recommended resolutions in the arguments section of this doc-string instead.